docs/ops.html

```diff
@@ -1247,7 +1247,168 @@ <h4 class="anchor-heading"><a id="prodconfig" class="anchor-link"></a><a href="#
+config.providers.envVarProvider.class=org.apache.kafka.common.config.provider.EnvVarConfigProvider</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="file_config_provider" class="anchor-link"></a>
+    <a href="#file_config_provider">FileConfigProvider</a>
+  </h4>
+  
+  <p>The <code>FileConfigProvider</code> retrieves configuration data from a single properties file.</p>
+  <p>You specify the path to the properties file using the <code>file</code> parameter.</p>
+  <p>This provider is useful for loading configuration data from mounted files.</p>
+  
+  <p><b>Example <code>FileConfigProvider</code> configuration</b></p>
+  <pre><code>config.providers=fileProvider
+config.providers.fileProvider.class=org.apache.kafka.common.config.provider.FileConfigProvider
+config.providers.fileProvider.param.file=/path/to/config.properties</code></pre>
+  
+  <h4 class="anchor-heading">
+    <a id="ref_config_provider" class="anchor-link"></a>
+    <a href="#ref_config_provider">Referencing configuration providers</a>
+  </h4>
+  <p>
+    To reference a property value supplied by a configuration provider, use the correct placeholder syntax:
+  </p>
+  <dl>
+    <dt><b><code>DirectoryConfigProvider</code></b></dt>
+    <dd>
+      <p><code>${dirProvider:&lt;path_to_file&gt;:&lt;file_name&gt;}</code></p>
+    </dd>
+    <dt><code><b>EnvVarConfigProvider</code></b></dt>
+    <dd>
+      <p><code>${envVarProvider:&lt;enVar_name&gt;}</code></p>
+    </dd>
+    
+    <dt><b><code>FileConfigProvider</code></b></dt>
+    <dd>
+      <p><code>${fileProvider:&lt;path_and_filename&gt;:&lt;property&gt;}</code></p>
+    </dd>
+  </dl>
+  <p>
+    Here’s an example that uses a file configuration provider with Kafka Connect to provide authentication credentials to a database for a connector.
+  </p>
+  <p>
+    First, create a <code>connector.properties</code> configuration file with the following credentials:
+  </p>
+  <p><b>Example configuration file</b></p>
+<pre><code>dbUsername=my-username
+dbPassword=my-password</code></pre>
+  <p><em>NOTE: Ensure the appropriate permissions are set for securely accessing the properties file.
+    For additional security, consider encrypting sensitive information using a secret and implementing decryption in your application at runtime.
+  </em></p>
+  <p>
+    Specify a <code>FileConfigProvider</code> in the Kafka Connect configuration:
+  </p>
+  
+  <p><b>Example Kafka Connect configuration with an <code>FileConfigProvider</code></b></p>
+  <pre><code>config.providers=fileProvider
+config.providers.fileProvider.class=org.apache.kafka.common.config.provider.FileConfigProvider
+config.providers.fileProvider.config.path=/path/to/connector-properties</code></pre>
+  
+  <p>Next, reference the properties from the file in the connector configuration.</p>
+  <p><b>Example connector configuration referencing file properties</b></p>
+  <pre><code>database.user=${file:/path/to/connector.properties:dbUsername}
```

Comment:
In this example, `file` should be `fileProvider` as the first part of the configuration provider magic string is the alias given to the provider.